wallet-integration/guide.mdx

```diff
@@ -30,24 +30,108 @@ To achieve this goal, you will need to do the following:
+
+This approach requires adding the `webNavigation` permission to your `manifest.json` file:
+
+```json
+{
+  "permissions": ["webNavigation"]
+}
+```
+
+<Warning>
+
+If your extension does not already have this permission, this will trigger a new
+permissions request when the user installs your extension to "Read your browsing history".
+
+Alternatively, you can make this permission optional by adding the
+`"optional_permissions": ["webNavigation"]` key to your `manifest.json` file instead.
+But this will require you to add additional logic to your extension to request this
+permission at the appropriate time (ex. in a settings page, or via a Call to Action
+prompt to opt-in to protection). You will also need to handle the case where the user
+denies the permission request.
+
+</Warning>
+
+Then you can listen for the `onBeforeNavigate` event in your Background script:
 
 ```ts
 import browser from "webextension-polyfill";
 
 browser.webNavigation.onBeforeNavigate.addListener(async function (details) {
-  //do stuff
+  // Filter out subframes and any navigation events that don't have a URL
+  if (details.frameId !== 0 || !details.url) {
+    return;
+  }
+
+  // ...
+});
+```
+
+### Option 2: Use the `browser.tabs.onUpdated` API
+
+This approach requires adding the `tabs` permission to your `manifest.json` file
+and requesting the `"<all_urls>"` host permission:
+
+```json
+{
+  "permissions": ["tabs"],
+  "host_permissions": ["<all_urls>"]
+}
+```
+
+<Warning>
+
+If your extension does not already have this permission, this will trigger a new
+permissions request when the user installs your extension to "Read your browsing history".
+
+Alternatively, you can make this permission optional by adding the
+`"optional_permissions": ["tabs"]` key to your `manifest.json` file, but this
+will require you to add additional logic to your extension to request this
+permission at the appropriate time (ex. in a settings page, or via a Call to Action
+prompt). You will also need to handle the case where the user denies the permission request.
+
+</Warning>
+
+Then you can listen for the `onUpdated` event in your Background script:
+
+```ts
+import browser from "webextension-polyfill";
+
+browser.tabs.onUpdated.addListener(async function (tabId, changeInfo, tab) {
+  // Filter out any tab change events that aren't complete, aren't for the
+  // active tab, or tabs that don't have a URL
+  if (changeInfo.status !== "complete" || !tab.active || !tab.url) {
+    return;
+  }
+
+  // ...
 });
 ```
 
-This code would typically go into your `Background` file.
+### Option 3: Get the current URL from a content script
+
+If you're already using a content script in your extension, you can get the current
+URL from the `window.location` object without needing to add any additional permissions.
 
-![Background file](https://user-images.githubusercontent.com/10101970/219277025-1a02f736-dbf0-4f63-84f6-611c9a502011.png)
+```ts
+const url = window.location.href;
```

Comment:
Can we add any suggestions on how to send this over to the extension? Non blocking but might be good for a follow up. 

This could also be a good spot to use the ChainPatrol sdk message passing

Reply:
Added some generic code snippets. The SDK does not currently have an event for this, and I want to keep this guide generic enough that even non-browser extension wallets could port over the code, so I've purposely omitted SDK code. I think I'll write up a separate version of the guide for the SDK at some point, but I don't want to do that as part of this ticket